test/shortwave_radiation.jl

```diff
@@ -0,0 +1,149 @@
+            @test column.tend_t_rsw ≈ [11.947794979452055, 27.611640775148402, 42.565299927060124, 40.46336031165737]
+        end
+    end
+    @testset "shortwave_radiation!" begin
+        @testset for NF in (Float32, Float64)
+            _, diagn, model = SpeedyWeather.initialize_speedy(NF, model = :primitive, nlev = 4)
+            nlev = diagn.nlev
+
+            column = ColumnVariables{NF}(nlev = nlev)
+
+            # 1. Set variables for sol_oz
+            column.tyear = 0.5
+            column.lat = 89.
+
+            # 2. Compute sat_vap_pres and dry_static_energy
+            # and set remainig varables for cloud
+            column.pres = 1. # nomalised
+            column.temp .= [208.40541, 219.8126 , 249.25502, 276.14264]
+            column.humid .= [0., 0.03124937, 0.9748285, 6.7846994] # g/kg
+            column.geopot .= [241932.19, 117422.14, 54618.79, 7626.5884]
+            SpeedyWeather.get_thermodynamics!(column, model)
+
+            # column.sat_vap_pres = [0.005265206274688095, 0.02494611200040165, 0.7012750147882735, 7.56823828640608]
+            # column.dry_static_energy = [451171.22164, 338113.9904, 304870.83008, 284873.79896]
+            # column.rel_hum = [0.0, 1.2526749659224194, 1.3900801817305823, 0.8964701087948754]
+            # column.grad_dry_static_energy = 0.4255393541723314
+
+            column.precip_convection = 1
+            column.precip_large_scale = 1
+            column.cloud_top = 1
+            column.fmask = 1.
+
+            # Set variables for radsw
+            # FIXME: pres is overloaded and will need to be
+            # fixed in other functions
+            column.norm_pres = column.pres
+            column.albsfc = 0.5
+
+            SpeedyWeather.shortwave_radiation!(column, model)
+
+            # Results need to be close to those computed by radsw but
+            # are not the same as precision is maintained between functions
+            @test isapprox(column.ssrd, 385.7997293028816, rtol=0.1)
+            @test isapprox(column.ssr, 192.8998646514408, rtol=0.1)
+            @test isapprox(column.tsr, 315.10016, rtol=0.1)
+            @test isapprox(column.tau2, [
+                0.9526258427031802 0.3788324531779398 1.0 1.0;
+                0.9286716429916902 0.2276374390970355 0.994618802250959 0.6801722651144487;
+                0.02148265489609933 0.12596241035550795 0.7900788064253661 4.906182803130207e-8;
+                0.9287847242363858 0.22819245387064319 0.31050206404533354 5.234705430612321e-37
+                ], rtol=0.1)
+            @test isapprox(column.tend_t_rsw, [
+                11.947794979452055, 27.611640775148402, 42.565299927060124, 40.46336031165737
+                ], rtol=0.1)
+            # Range is generally order 10^-5
+            @test column.temp_tend ≈ [7.189150408440864e-6, 1.2141337616807728e-5, 1.3196339560843619e-5, 1.5994200814647754e-5]
```

Comment:
Where do all of the numbers in these tests come from? From the Fortran Speedy functions?